cmd/jujud/agent/machine/manifolds_test.go

```diff
@@ -223,6 +227,31 @@ func (*ManifoldsSuite) TestSingularGuardsUsed(c *gc.C) {
 	}
 }
 
+func (*ManifoldsSuite) TestAPICallerNonRecoverableErrorHandling(c *gc.C) {
+	tmpDir, err := ioutil.TempDir("", "agent-data-dir")
```

Comment:
This should be c.Mkdir() as that will be cleaned up by the test infrastructure.

Reply:
Pushed https://github.com/juju/juju/pull/10193/commits/96777d5d16ae76ec1c4279a71cbb0bea2625aceb to switch to c.MkDir()